### PR TITLE
[mlrun] Support chief/worker deployment

### PR DESCRIPTION
**Pull request by Hedingber**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description:
Related to https://github.com/mlrun/mlrun/pull/1960
The `app.kubernetes.io/sub-component` (`chief/worker`) label I added is not from the [k8s recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) but something I invented
I had to keep the `app.kubernetes.io/component` set to `api` for both the chief and the workers so I would be able to identify (service selector) both the chief and the workers from the API k8s svc (but not the UI) and in addition to have a way to identify only the chief API for the chief api k8s svc
Unfortunately k8s svc does not support [set based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) which would made this much easier

### CODE REVIEW

File: stable/mlrun/values.yaml [hunk 1]
```diff
@@ -8,11 +8,17 @@ api:
 
   logLevel: INFO
 
-  replicaCount: 1
+  chief:
+    fullnameOverride:
```
Comment (Tankilevitch) [inline stable/mlrun/values.yaml:12]:
```suggestion
    fullNameOverride:
```

Comment (Tankilevitch) [reply stable/mlrun/values.yaml:12]: Is it on purpose that only the chief api has the ability to override its full name?

File: stable/mlrun/templates/api-worker-deployment.yaml [hunk 2]
```diff
@@ -1,18 +1,19 @@
+{{- if .Values.api.worker.minReplicaCount -}}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "mlrun.api.fullname" . }}
```
Comment (Tankilevitch) [inline stable/mlrun/templates/api-worker-deployment.yaml]: I think maybe we should inherit the worker name from the chief deployment, what do you think?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: stable/mlrun/values.yaml (hunk 1 above)
Comment (Hedingber) [reply stable/mlrun/values.yaml:12]: I agree full and name are different words therefore your suggestion makes sense
BUT, that `fullnameOverride` is something that you get in the [initial starter chart template](https://helm.sh/docs/chart_template_guide/getting_started/#a-starter-chart) therefore if you'll look around on charts (ours, and in the open source) you'll see it is like that everywhere, so prefer to leave it as is

File: stable/mlrun/templates/api-chief-ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{{- if .Values.api.chief.ingress.enabled -}}
+{{- $fullName := include "mlrun.api.chief.fullname" . -}}
+{{- $svcPort := .Values.api.chief.service.port -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (Tankilevitch) [inline stable/mlrun/templates/api-chief-ingress.yaml:4]: 😍

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**